sdk/attestation/azure-security-attestation/src/main/java/com/azure/security/attestation/AttestationAsyncClient.java

```diff
@@ -8,43 +8,110 @@
 import com.azure.core.annotation.ServiceClient;
 import com.azure.core.annotation.ServiceMethod;
 import com.azure.core.http.rest.Response;
+import com.azure.core.util.Context;
+import com.azure.security.attestation.implementation.AttestationClientImpl;
 import com.azure.security.attestation.implementation.AttestationsImpl;
+import com.azure.security.attestation.implementation.MetadataConfigurationsImpl;
+import com.azure.security.attestation.implementation.SigningCertificatesImpl;
 import com.azure.security.attestation.models.AttestOpenEnclaveRequest;
 import com.azure.security.attestation.models.AttestSgxEnclaveRequest;
 import com.azure.security.attestation.models.AttestationResponse;
+import com.azure.security.attestation.models.AttestationSigner;
 import com.azure.security.attestation.models.CloudErrorException;
 import reactor.core.publisher.Mono;
 
 import java.nio.charset.StandardCharsets;
 import java.util.Objects;
 
+import static com.azure.core.util.FluxUtil.withContext;
+
 /** Initializes a new instance of the asynchronous AzureAttestationRestClient type. */
-@ServiceClient(builder = AttestationClientBuilder.class, isAsync = true)
+@ServiceClient(builder = AttestationAsyncClientBuilder.class, isAsync = true)
 public final class AttestationAsyncClient {
-    private final AttestationsImpl serviceClient;
+    private final AttestationsImpl attestImpl;
+    private final MetadataConfigurationsImpl metadataImpl;
+    private final SigningCertificatesImpl signerImpl;
 
     /**
      * Initializes an instance of Attestations client.
      *
-     * @param serviceClient the service client implementation.
+     * @param clientImpl the service client implementation.
+     */
+    AttestationAsyncClient(AttestationClientImpl clientImpl) {
+        this.attestImpl = clientImpl.getAttestations();
+        this.metadataImpl = clientImpl.getMetadataConfigurations();
+        this.signerImpl = clientImpl.getSigningCertificates();
+    }
+
+    /**
+     * Retrieves metadata about the attestation signing keys in use by the attestation service.
+     *
+     * @param context - Context for the operation
+     * @throws CloudErrorException thrown if the request is rejected by server.
+     * @throws RuntimeException all other wrapped checked exceptions if the request fails to be sent.
+     * @return any object.
+     */
+//    @ServiceMethod(returns = ReturnType.SINGLE)
```

Comment:
Why is this not a service method?

Reply:
Because checkStyle rejects it if it is a service method :(.

Apparently you can't have context parameters on a ServiceMethod. I have no idea why, but this way passes.

Reply:
You need a service method for getOpenIdMetadataWithResponse without context as argument.

Reply:
Context in async methods come from the reactor context --> so we do not take this as an argument from the customer.

you could do something like
@ServiceMethod(returns = ReturnType.SINGLE)
    public Mono<Response<Object>> getOpenIdMetadataWithResponse() {) and a non-public method  Mono<Response<Object>> getOpenIdMetadataWithResponse(Context context) {) and wire XXX and XXXWithResponse to the non-public XXXWithResponse method.